app/controllers/review_item/reject.js

```diff
@@ -46,120 +46,159 @@ export default Ember.Controller.extend({
+      "rejection_reason",
+      this.get("selectedId")
+    );
+    rejectProperties.state_event = "reject";
+    rejectProperties.id = this.get("itemId");
+
+    rejectProperties.offer = this.get("offer.model");
+    rejectProperties.packageType = this.store.peekRecord(
+      "packageType",
+      this.get("itemTypeId")
+    );
+    return rejectProperties;
+  },
 
+  actions: {
     setRejectOption() {
       this.set("selectedId", "-1");
     },
 
+    setMessage(message) {
+      this.set("rejectMsg", message);
+    },
+
     rejectItem() {
-      if(this.get("itemId") && this.cannotSave()){
+      if (this.get("itemId") && this.cannotSave()) {
         return false;
       }
-      var selectedReason = this.get('selectedId');
-      var rejectProperties = this.getProperties('rejectReason');
-      if(!this.rejectValidation(selectedReason, rejectProperties)){
+      var selectedReason = this.get("selectedId");
+      var rejectProperties = this.rejectProperties();
+
+      if (!this.rejectValidation(selectedReason, rejectProperties)) {
         return false;
       }
 
-
-      if(selectedReason !== "-1") {
+      if (selectedReason !== "-1") {
         rejectProperties.rejectReason = null;
-        this.set('rejectReason', null);
+        this.set("rejectReason", null);
       }
 
       var offer = this.get("offer.model");
 
       var saveItem = () => {
-
-
-        var loadingView = getOwner(this).lookup('component:loading').append();
-        rejectProperties.rejectionReason = this.store.peekRecord('rejection_reason', selectedReason);
-        rejectProperties.state_event = 'reject';
-        rejectProperties.id = this.get('itemId');
-
-        rejectProperties.offer = offer;
-        rejectProperties.packageType = this.store.peekRecord('packageType', this.get('itemTypeId'));
-
+        var loadingView = getOwner(this)
+          .lookup("component:loading")
+          .append();
         var item = this.store.peekRecord("item", this.get("itemId"));
         item.setProperties(rejectProperties);
 
         // Save changes to Item
-        item.save()
+        item
+          .save()
           .then(() => {
-            this.transitionToRoute('review_offer.items');
-            this.get("reviewOfferController").set("displayCompleteReviewPopup", offer.get("allItemsReviewed") && offer.get("isUnderReview"));
+            this.transitionToRoute("review_offer.items");
+            this.get("reviewOfferController").set(
+              "displayCompleteReviewPopup",
+              offer.get("allItemsReviewed") && offer.get("isUnderReview")
+            );
           })
           .catch(error => {
-            if(item) {
+            if (item) {
               item.rollback();
             }
-            if (error.errors instanceof Array &&
-              error.errors.filter(e => !!e["requires_gogovan_cancellation"]).length > 0) {
-              return this.transitionToRoute('offer.cancel_gogovan', offer);
+            if (
```

Comment:
I know these are old codes but if you would like to you can clear these `if` conditions with some meaningful name to make it more readable :)